lib/codecs/src/encoding/format/csv.rs

```diff
@@ -59,16 +134,78 @@ pub struct CsvSerializerOptions {
     pub fields: Vec<ConfigTargetPath>,
 }
 
+const fn default_delimiter() -> u8 {
+    b','
+}
+
+const fn default_escape() -> u8 {
+    b'"'
+}
+
+const fn default_double_quote() -> bool {
+    true
+}
+
+const fn default_capacity() -> usize {
+    8 * (1 << 10)
+}
+
+impl Default for CsvSerializerOptions {
+    fn default() -> Self {
+        Self {
+            delimiter: default_delimiter(),
+            double_quote: default_double_quote(),
+            escape: default_escape(),
+            quote: default_escape(),
+            quote_style: QuoteStyle::default(),
+            capacity: default_capacity(),
+            fields: Vec::new(),
+        }
+    }
+}
+
+impl CsvSerializerOptions {
+    fn csv_quote_style(&self) -> csv_core::QuoteStyle {
+        match self.quote_style {
+            QuoteStyle::Always => csv_core::QuoteStyle::Always,
+            QuoteStyle::Necessary => csv_core::QuoteStyle::Necessary,
+            QuoteStyle::NonNumeric => csv_core::QuoteStyle::NonNumeric,
+            QuoteStyle::Never => csv_core::QuoteStyle::Never,
+        }
+    }
+}
+
 /// Serializer that converts an `Event` to bytes using the CSV format.
 #[derive(Debug, Clone)]
 pub struct CsvSerializer {
+    wtr: csv_core::Writer,
```

Comment:
Please avoid abbreviated variable names. 

Also, this introduced a [clippy error](https://github.com/vectordotdev/vector/actions/runs/5929932270/job/16080197647?pr=18320). See link for details and potential fixes.

Reply:
Agreed. I tried to Box the writer. Again I am not sure of potential performance impacts.